pkg/controller/compliancescan/scan.go

```diff
@@ -47,6 +56,32 @@ func (r *ReconcileComplianceScan) createScanPods(instance *compv1alpha1.Complian
 	return nil
 }
 
+func (r *ReconcileComplianceScan) createPlatformScanPod(instance *compv1alpha1.ComplianceScan, logger logr.Logger) error {
+	logger.Info("Creating a Platform scan pod")
+	pod := newPlatformScanPod(instance, logger)
+	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
+		log.Error(err, "Failed to set pod ownership", "pod", pod)
+		return err
+	}
+
+	err := r.client.Create(context.TODO(), pod)
+	if errors.IsAlreadyExists(err) {
+		logger.Info("Pod already exists. This is fine.", "pod", pod)
+	} else if err != nil {
+		log.Error(err, "Failed to launch a pod", "pod", pod)
+		return err
+	} else {
+		logger.Info("Launched a pod", "pod", pod)
+	}
+
+	// make sure the instance is updated with the node-pod labels
+	// TODO: Is this needed for node-pod labels or ownerref?
```

Comment:
Why are node-pod-labels for a platform scan needed? About the ownerref, I wouldn't update the pod for it, either it doesn't exist in which case the create does all the work or the pod exists and imo the function shouldn't then touch it.

Reply:
removed this part